lib/matplotlib/tests/test_axes.py

```diff
@@ -3635,6 +3635,13 @@ def test_bar_negative_width():
         assert_equal(b._width, 1)
         assert_equal(b._height, indx + 1)
 
+@image_comparison(baseline_images=['square_plot'], extensions=['png'])
+def test_square_plot():
```

Comment:
This can probably be done with out an image test, just check that the aspect is equal and the ranges are the same.